Standards/scs-0101-w1-entropy-implementation-testing.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "SCS Entropy: Implementation and Testing Notes"
+type: Supplement
+status: Draft
+supplements:
+  - scs-0101-v1-entropy.md
+---
+
+## Implementation
+
+We presume that almost nothing has to be done (or indeed can be done), as
+long as the CPUs and VM images are reasonably recent; only the flavor and
+image attributes have to be set:
+
+- flavor: `hw_rng:allowed=True` ,
+- image: `hw_rng_model: virtio` .
+
+## Automated Tests
+
+### Images Sample
+
+Some checks need to be performed on a live instance. For these checks, it is
+necessary to choose a sample of VM images to test on.
+
+For the time being, the sample MUST contain the very first public image reported
+by OpenStack. This may be extended in the future.
```

Comment:
I have no idea what kind of image this points to. Is this like a default image? As I don't know what it points to, it may be perfectly understandable, but for me, some more clarification or different wording would be excellent.

Reply:
The script just fetches the list of images from Glance, and it picks the very first one. There is no telling what image this will be. And I think this is bad, because this image might not be representative at all. I just don't have a better idea. On the one hand we need to narrow down this image selection to make it more predictable; one the other hand, SCS doesn't require any image to be present except maybe Ubunut, and I don't want to hard-code this here (to avoid duplication with the images standard),  Suggestions welcome!

Reply:
One more thing: we could test EVERY image; maybe this is even necessary to be really safe. But this seems a bit overkill to me. We mainly want to make sure that the guest kernel has access to the CPU instructions. And for everything else, you can always use your own image. (The images standard goes into this direction.)

Reply:
I think we need to narrow down the images for this test. Case in point:
```
DEBUG: Connecting to cloud 'wave1-compliance'
...
DEBUG: Creating instance of image 'Cirros 0.6.2' using flavor 'SCS-1L-1-5'
DEBUG: Server '_scs-0101-server' ('78b8b449-43d5-4c8c-b5db-1203c4ed0e93') has been created
DEBUG: Error running 'apt-get': sh: apt-get: not found
DEBUG: Error running 'dnf': sudo: dnf: command not found
DEBUG: Error running 'yum': sudo: yum: command not found
DEBUG: Error running 'pacman': sudo: pacman: command not found
DEBUG: No package manager worked; proceeding anyway as rng-utils might be present nonetheless.
ERROR: VM 'Cirros 0.6.2': failed to determine fips failures
DEBUG: stderr following:
sh: rngtest: not found

DEBUG: Deleting server '_scs-0101-server'
DEBUG: Total critical / error / info: 0 / 1 / 1
```

Reply:
Imho it is enough to use one of the images (that will always be there) for the test.

Reply:
I changed it so that it tries to use a Debian derivative if possible. Et voila, it works much better than Cirros:

```
DEBUG: Connecting to cloud 'wave1-compliance'
DEBUG: Selected image: Debian 12
DEBUG: Checking images and flavors for recommended attributes
WARNING: Flavor 'SCS-16V-64-500s_GNa-14h' missing recommended attributes: hw_rng:allowed=True; additionally, missing optional attributes: hw_rng:rate_bytes, hw_rng:rate_period
DEBUG: Checking dynamic instance properties
DEBUG: Using external network 68a806ee-4eb8-4b50-ae49-c06bde9baf06.
DEBUG: Creating instance of image 'Debian 12' using flavor 'SCS-1V-16-50'
DEBUG: Server 'scs-0101-server' ('599071c1-b94b-47db-a610-dfb789aaf3f1') has been created
INFO: VM 'Debian 12' doesn't provide the recommended service rngd
DEBUG: Deleting server 'scs-0101-server'
DEBUG: Total critical / error / info: 0 / 0 / 1
```